sources/core/Stride.Core.AssemblyProcessor/Stride.Core.AssemblyProcessor.csproj

```diff
@@ -76,38 +76,49 @@
 
     <PackageReference Include="Microsoft.Build.Utilities.Core" Version="17.3.2" ExcludeAssets="runtime" />
   </ItemGroup>
-  <!-- Generate hash file for assembly -->
+  
+  <!-- Define a custom task for writing text to a file -->
   <UsingTask
-  TaskName="WriteAllText"
-  TaskFactory="RoslynCodeTaskFactory"
-  AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll" >
+    TaskName="WriteAllText"
+    TaskFactory="RoslynCodeTaskFactory"
+    AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll">
     <ParameterGroup>
-      <Path ParameterType="System.String" />
-      <Contents ParameterType="System.String" />
+      <Path ParameterType="System.String"/>
+      <Contents ParameterType="System.String"/>
     </ParameterGroup>
     <Task>
       <Using Namespace="System"/>
       <Using Namespace="System.IO"/>
       <Code Type="Fragment" Language="cs">
         <![CDATA[
-File.WriteAllText(Path, Contents);
-]]>
+          File.WriteAllText(Path, Contents);
+        ]]>
       </Code>
     </Task>
   </UsingTask>
-  <!--<UsingTask TaskName="Microsoft.Build.Tasks.GetFileHash" AssemblyName="Microsoft.Build.Tasks.Core, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"/>
-  <UsingTask TaskName="Microsoft.Build.Tasks.VerifyFileHash" AssemblyName="Microsoft.Build.Tasks.Core, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"/>-->
 
+  <ItemGroup>
+    <BuiltItems Include="$(TargetDir)*" />
+  </ItemGroup>
 
-  <Target Name="GenerateHash" AfterTargets="PostBuildEvent">
-    <Exec Command="xcopy /y $(TargetDir)*.* $(SolutionDir)..\deps\AssemblyProcessor\$(TargetFramework)\
-cd $(SolutionDir)..\deps\AssemblyProcessor\$(TargetFramework)\
-$(ILRepack) Stride.Core.AssemblyProcessor$(TargetExt) Mono.Cecil.dll Mono.Cecil.Mdb.dll Mono.Cecil.Pdb.dll Mono.Cecil.Rocks.dll Mono.Options.dll /out:Stride.Core.AssemblyProcessor.Packed$(TargetExt)"/>
+  <!-- Copy the built items to the deps folder -->
+  <Target Name="CopyFiles" AfterTargets="PostBuildEvent">
+    <Copy
+      SourceFiles="@(BuiltItems)"
+      DestinationFolder="$(SolutionDir)..\deps\AssemblyProcessor\$(TargetFramework)"
+    />
+  </Target>
+  
+  <Target Name="GenerateHash" AfterTargets="CopyFiles">
+    <!-- Repack the assemblies into a single file -->
+    <Exec Command="cd $(SolutionDir)..\deps\AssemblyProcessor\$(TargetFramework)
+    $(ILRepack) Stride.Core.AssemblyProcessor$(TargetExt) Mono.Cecil.* Mono.Options.* /out:Stride.Core.AssemblyProcessor.Packed$(TargetExt)"/>
```

Comment:
This line fails to build on my end, here's an excerpt of the trace:
```
1>INFO: ------------- IL Repack Arguments -------------
1>/out:Stride.Core.AssemblyProcessor.Packed.dll  Stride.Core.AssemblyProcessor.dll Mono.Cecil.* Mono.Options.*
1>-----------------------------------------------
1>INFO: Adding assembly for merge: Stride.Core.AssemblyProcessor.dll
1>J:\stride\sources\core\Stride.Core\Storage\ObjectIdBuilder.cs(331,26,331,58): warning CS1030: #warning: 'PERF: Do not copy byte-for-byte.'
1>J:\stride\sources\core\Stride.Core\Storage\ObjectIdBuilder.cs(359,26,359,58): warning CS1030: #warning: 'PERF: Do not copy byte-for-byte.'
1>J:\stride\sources\core\Stride.Core.AssemblyProcessor\SerializationProcessor.cs(580,13,580,58): warning CS0618: 'ObjectIdBuilder.Write(byte*, int)' is obsolete: 'Use Write(ReadOnlySpan<byte>)'
1>Stride.Core.AssemblyProcessor -> J:\stride\sources\core\Stride.Core.AssemblyProcessor\bin\Debug\net6.0\Stride.Core.AssemblyProcessor.dll
1>INFO: IL Repack - Version 2.0.18
1>INFO: ------------- IL Repack Arguments -------------
1>/out:Stride.Core.AssemblyProcessor.Packed.dll  Stride.Core.AssemblyProcessor.dll Mono.Cecil.* Mono.Options.*
1>-----------------------------------------------
1>INFO: Failed to load debug information for Stride.Core.AssemblyProcessor.dll
1>INFO: Adding assembly for merge: Mono.Cecil.*
1>INFO: Adding assembly for merge: Stride.Core.AssemblyProcessor.dll
1>EXEC : error : Failed to load assembly Mono.Cecil.*
1>System.ArgumentException: Illegal characters in path.
1>   at System.Security.Permissions.FileIOPermission.EmulateFileIOPermissionChecks(String fullPath)
1>   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
1>   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
1>   at Mono.Cecil.ModuleDefinition.GetFileStream(String fileName, FileMode mode, FileAccess access, FileShare share)
1>   at Mono.Cecil.ModuleDefinition.ReadModule(String fileName, ReaderParameters parameters)
1>   at ILRepacking.ILRepack.ReadInputAssembly(String assembly, Boolean isPrimary)
1>   at ILRepacking.ILRepack.ReadInputAssemblies()
1>   at ILRepacking.ILRepack.Repack()
1>   at ILRepacking.Application.Main(String[] args)
```
Swapping the asterisk in `Mono.Cecil.*` and `Mono.Options.*` with `dll` fixes it.

Those are c# libraries right, aren't those files guaranteed to be .dlls regardless of the os ?

Reply:
As an addition, we need to test this if `SolutionDir` contains a space.
We likely need to add `&quot;` (MSBuild escaped quotation) around the path like this:
cd &quot;$(SolutionDir)..\deps\AssemblyProcessor\$(TargetFramework)&quot;

Reply:
I'll test this out tomorrow.